src/path/fs.rs

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2018 The predicates-rs Project Developers.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::fmt;
+use std::path;
+
+use path::{FileContent, FileTypePredicate};
+use Predicate;
+
+/// Predicate that compares file matches
+#[derive(Clone, Debug)]
+pub struct FileSystemEntryPredicate {
```

Comment:
Should we make this only work with files?  We can handle dirs like we handle symlinks or non-existent files

If we keep it handling dirs, maybe shorten the name to `EntryPredicate`?